### DISCUSSION TITLE
Input: Maximum number of characters

### DISCUSSION THREAD

**Discussion by senaoezdogan** [cypress-io/cypress#21626] [Component Testing]
I have to test the maximum number of characters for an input. For example, the first name can only have 30 characters. After more than 30 characters, nothing more can be typed. My idea was to enter a first name with 40 characters and to see if the input length is only 30 characters. How can I check the input length, or do you have other ideas, how to check?

**Comment by jjhelguero** [comment]
You can access the `length` of the text after you retrieve it from your input. You may also want to check your input does not match what you typed. 
```js
// use Cypress._.repeat() to type 40 characters
const moreThanMaxLength = Cypress._.repeat("asdf", 10);
cy.get("#fName").should("be.visible").type(moreThanMaxLength);
cy.get("#fName").should((text) => {
  const val = text[0].value;
  expect(val.length).to.equal(30);
  expect(val).to.not.equal(moreThanMaxLength);
});
```

Here is the [example](https://github.com/jjhelguero/cypress-examples/blob/main/cypress/integration/inputMaxLength.md) I used.

**Reply by senaoezdogan** [reply]
Thank you, it works fine!
